### PR TITLE
Port to futures 0.3

### PR DESCRIPTION
**Pull request by sdroege**
While this requires nightly, the API is now stabilized and will become part of Rust 1.36. IMHO we should merge this now so that it can all get wider testing.

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
Looks good to me, thanks! And yes, the sooner the better.

**Comment by sdroege** [comment]
Great, thanks. Waiting for @EPashkin's opinion and the "OK" for all the other PRs and then let's get merging

**Comment by sdroege** [comment]
> and then let's get merging

Note that I have to update all PRs before merging apart from glib and gir once those two are merged.

**Comment by EPashkin** [comment]
@sdroege Big thanks.
As we have some fixes for previous release maybe better do release first?
Then we have 3 month until stable 1.36 for testing.

**Comment by sdroege** [comment]
The idea would be to include it in this release already so people actually start testing. Almost nobody is using git :)

**Comment by EPashkin** [comment]
Ok

**Comment by sdroege** [comment]
Ok, then I'll merge this and update the other ones in a bit.

File: src/main_context_futures.rs [hunk 1]
```diff
@@ -283,17 +314,17 @@ impl MainContext {
...
+                future::ready(())
             });
 
-            let f: *mut Future<Item = (), Error = Never> = Box::into_raw(Box::new(f));
-            // XXX: Transmute to get a 'static lifetime here, super unsafe
-            let f: *mut (Future<Item = (), Error = Never> + 'static) = mem::transmute(f);
-            let f: Box<Future<Item = (), Error = Never> + 'static> = Box::from_raw(f);
+            // Super-unsafe: We transmute here to get rid of the 'static lifetime
+            let f = LocalFutureObj::new(Box::new(f));
+            let f: (LocalFutureObj<'static, ()>) = mem::transmute(f);
```
Comment (EPashkin) [inline src/main_context_futures.rs]: IMHO comment now has wrong part "get rid" instead just "get"

**Pull request merged**